.github/workflows/render_build_deploy.yaml

```diff
@@ -36,6 +36,7 @@ jobs:
         run: |
           git config --local user.email "actions@github.com"
           git config --local user.name "GitHub Actions"
+          git checkout full-pdf
```

Comment:
I wonder if we should set the branch in the `env` instead of using `git checkout`. 

This isn't a Quarto example, but the idea seems to be similar (build something from main, and push it elsewhere):  [https://gist.github.com/umanghome/30527c68cd605dfef8702a832ff14fd5](https://gist.github.com/umanghome/30527c68cd605dfef8702a832ff14fd5)

Something like:
```
env:
    github_token: ${{ secrets.GITHUB_TOKEN }}
    REPO: self
    BRANCH: full-pdf
```
This actually might mean the `gh-pages` branch could work here instead - there was an error (I think?) when we tried using `git checkout` in actions with the `gh-pages` branch before. If we set gh-pages in the env as the branch I think it might work?

Reply:
So, remove git checkout, then add the new code to the `env:` section of the render job?

Reply:
Actually, I'm not sure. I'm playing around with actions at the moment (https://github.com/nrennie/github-actions-playground/) and neither of these approaches have worked. I'll have another dig around to see if I can get a minimal version working first.

Reply:
This worked: https://github.com/nrennie/github-actions-playground/blob/main/.github/workflows/render_build_deploy.yaml 

If you edit the bit at the bottom to add in a separate github action step for pushing to a different branch (and the fetch step nearer the top), it should work. This might also work with the branch to push to as gh-pages but I haven't checked.

Reply:
I tried it with gh-pages, but it threw up an error. Switched to full-pdf and it worked, but the rendered PDF was still pushed to main before all commits (not just the new pdf) were then pushed to the full-pdf branch.

So, unfortunately, this doesn't solve the problem of stopping the rendered PDF being pushed to main.

Reply:
Updated version using a github pages deploy action but pushing to a different branch *seems* to work (doesn't push to main at least...): https://github.com/nrennie/github-actions-playground/blob/main/.github/workflows/render_build_deploy.yaml